### PR TITLE
[Ready] *smacks table* Objection!

### PR DESCRIPTION
**Pull request by Urumasi**
## About The Pull Request

Direct extension of #57269

After a [suggestion](https://i.imgur.com/yaeWSnE.png), I decided to make slamming a table while wearing an attorney's badge yell "Objection!!"
Also makes slamming tables a signal

## Why It's Good For The Game

Objection!

## Changelog
:cl:
add: Lawyers can now object by slamming the table.
code: Table slamming is now handled using signals.
/:cl:

### CODE REVIEW

**Comment by Urumasi** [comment]
I'm closing this, going to rework it into action buttons

**Comment by itseasytosee** [comment]
Don't make it an action button. This PR is perfect. We don't need more screen clutter. There is no need to shove a feature like this in someone's face.

**Comment by Urumasi** [comment]
Re-opening after trying to find a copyright-free Objection! sound to perhaps play if you're in the Courtroom, but I decided this PR is good enough on it's own

File: code/game/objects/items/hand_items.dm [hunk 1]
```diff
@@ -151,6 +151,10 @@
 			if(istype(human_user.shoes, /obj/item/clothing/shoes/cowboy))
 				human_user.say(pick("Hot damn!", "Hoo-wee!", "Got-dang!"), spans = list(SPAN_YELL), forced=TRUE)
 				human_user.client?.give_award(/datum/award/achievement/misc/hot_damn, human_user)
+			else if(istype(human_user.w_uniform, /obj/item/clothing/under))
+				var/obj/item/clothing/under/uniform = human_user.w_uniform
+				if(istype(uniform.attached_accessory, /obj/item/clothing/accessory/lawyers_badge))
+					human_user.say("Objection!!", spans = list(SPAN_YELL), forced=TRUE)
```
Comment (Mothblocks) [inline code/game/objects/items/hand_items.dm:157]: I think now that we have more than one thing that cares about this, it should just be made into a signal. It's going to be more work for you, but it's cleaner.

Comment (Urumasi) [reply code/game/objects/items/hand_items.dm:157]: I'll look into it

File: code/__DEFINES/dcs/signals.dm [hunk 2]
```diff
@@ -653,6 +653,9 @@
 ///from base of mob/living/simple_animal/hostile/regalrat: (mob/living/simple_animal/hostile/regalrat/king)
 #define COMSIG_RAT_INTERACT "rat_interaction"
 
+///from /obj/item/slapper/attack_obj(obj/O, mob/living/user, params)
```
Comment (Mothblocks) [inline code/__DEFINES/dcs/signals.dm:656]: No `obj/O`--https://github.com/tgstation/tgstation/blob/master/.github/CONTRIBUTING.md#use-descriptive-and-obvious-names

File: code/game/objects/items/hand_items.dm [hunk 3]
```diff
@@ -146,15 +146,7 @@
...
-		if(ishuman(user))
-			var/mob/living/carbon/human/human_user = user
-			if(istype(human_user.shoes, /obj/item/clothing/shoes/cowboy))
-				human_user.say(pick("Hot damn!", "Hoo-wee!", "Got-dang!"), spans = list(SPAN_YELL), forced=TRUE)
-				human_user.client?.give_award(/datum/award/achievement/misc/hot_damn, human_user)
-			else if(istype(human_user.w_uniform, /obj/item/clothing/under))
-				var/obj/item/clothing/under/uniform = human_user.w_uniform
-				if(istype(uniform.attached_accessory, /obj/item/clothing/accessory/lawyers_badge))
-					human_user.say("Objection!!", spans = list(SPAN_YELL), forced=TRUE)
+		SEND_SIGNAL(user, COMSIG_LIVING_TABLE_SLAM, the_table)
```
Comment (Mothblocks) [inline code/game/objects/items/hand_items.dm:149]: Your documentation isn't correct--you only send the table, nothing else.

**Review by Mothblocks — CHANGES_REQUESTED** [review]

File: code/game/objects/items/hand_items.dm [hunk 4]
```diff
@@ -147,6 +147,7 @@
 			color = COLOR_GREEN
 		user.do_attack_animation(the_table)
 		SEND_SIGNAL(user, COMSIG_LIVING_TABLE_SLAM, user, the_table)
+		SEND_SIGNAL(the_table, COMSIG_LIVING_TABLE_SLAM, user, the_table)
```
Comment (Mothblocks) [inline code/game/objects/items/hand_items.dm:150]: This should be a different signal.

**Comment by 81Denton** [comment]
@Mothblocks Is this good to go?

**Comment by Mothblocks** [comment]
yeah, forgot about it sorry

**Pull request merged**